src/Model/AggregatesTrait.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atk4\Data\Model;
+
+use Atk4\Data\Model;
+
+/**
+ * Provides aggregation methods.
+ */
+trait AggregatesTrait
+{
+    /**
+     * @param array|object $seed
+     *
+     * @return Aggregate
+     *
+     * @see Aggregate::withAggregateField.
+     */
+    public function withAggregateField(string $name, $seed = []): Model
```

Comment:
should return `static`, eg. the `Aggregate` model wrapped in current Model to support hinting

also, I think this will fail with DB vendors as some require all non-unique fields to be specified to be somehow aggregated